mirgecom/steppers.py

```diff
@@ -59,6 +58,17 @@ def advance_state(rhs, timestepper, checkpoint, get_timestep,
         Time at which to start
     istep: int
         Step number from which to start
+    pre_step_callback
+        An optional user-defined function to be called before the timestepper
+        is called for that particular step. A non-zero return code from this
+        function indicates that this function should stop gracefully.
+        Examples of such functions include visualization, io, or restart
+        checkpoints.
+    post_step_callback
+        An optional user-defined function to be called after the timestepper
+        is called for that particular step. A non-zero return code from this
+        function indicates that this function should stop gracefully.
+        Examples of such functions include applying modal filtering or limiters.
```

Comment:
Can we get by with one callback?

Reply:
There are some callback routines that would you definitely want to apply after the timestepper though. Like modal filtering

Reply:
Maybe... ultimately the distinction only matters for the first (and possibly last) step though. I could be talked into believing that two is more correct.

Reply:
Having pre and post hooks available have been useful for me in other projects where we were coupling several solvers and other components together. Although this infrastructure is (currently) monolithic, and more flexible than the code in those other projects, my intuition tells me both would be useful.  

That said, at the moment if we just had the pre-hook, it would be fine.  We don't have a real use for the post-hook atm, and it is easy to add.  I'm ok with doing away with the post-hook if we'd rather not have it.

Reply:
Sure, let's keep both.